File: backend/src/main/java/com/plugga/backend/service/DeckCardService.java
package com.plugga.backend.service;

import com.plugga.backend.entity.DeckCard;

import java.util.List;

public interface DeckCardService {
    List<DeckCard> findAll();

    DeckCard findById(int id);

    List<DeckCard> findByDeckId(int id);

    List<DeckCard> findByCardId(int id);

    void save(DeckCard deckCard);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/service/DeckCardService.kt
package com.plugga.backend.service

import com.plugga.backend.entity.DeckCard
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface DeckCardService {
    fun findAll(pageable: Pageable): Page<DeckCard>

    fun findById(id: Int): DeckCard?

    fun findByDeckId(pageable: Pageable, id: Int): Page<DeckCard>

    fun findByCardId(pageable: Pageable, id: Int): Page<DeckCard>

    fun save(deckCard: DeckCard)

    fun deleteById(id: Int)
}
